src/test/java/seedu/address/model/ModelManagerTest.java

```diff
@@ -68,6 +71,66 @@ public void addAndGetHistoryCommand() {
         HistoryCommand historyCommand = historyCommands.get(0);
     }
 
+    @Test
+    public void testCommitAndUndoAddressBook() throws CommandException {
+        AddressBook addressBook1 = new AddressBook();
+        AddressBook addressBook2 = getTypicalAddressBook();
+
+        ModelManager model = new ModelManager(addressBook1, new UserPrefs());
+        model.commitAddressBook();
+        model.setAddressBook(addressBook2);
+        assertEquals(addressBook2, model.getAddressBook());
+
+        model.commitAddressBook();
+        model.undoAddressBook();
+        assertEquals(addressBook1, model.getAddressBook());
```

Comment:
I see! The ammendment in the ``equals()`` method in ``Address`` class enables the direct comparison between ``AddressBook`` and ``current`` in the ``VersionedAddressBook``!

Reply:
Yes! otherwise it will wrongly compare with `VAB.persons` with is obsolete (it exists in VAB but is not used)